lldb/unittests/SymbolFile/DWARF/SymbolFileDWARFTests.cpp

```diff
@@ -132,34 +134,33 @@ TEST_F(SymbolFileDWARFTests, ParseArangesWithMultipleTerminators) {
   // Set the big endian length correctly.
   const offset_t binary_data_size = sizeof(binary_data);
   binary_data[3] = (uint8_t)binary_data_size - 4;
-  DWARFDataExtractor data;
-  data.SetData(static_cast<const void *>(binary_data), sizeof binary_data,
-               lldb::ByteOrder::eByteOrderBig);
+  llvm::DWARFDataExtractor data(llvm::ArrayRef(binary_data),
+                                /*isLittleEndian=*/false, /*AddrSize=*/4);
   DWARFDebugArangeSet set;
   offset_t off = 0;
   llvm::Error error = set.extract(data, &off);
   // Multiple terminators are not fatal as they do appear in binaries.
   EXPECT_FALSE(bool(error));
   // Parser should read all terminators to the end of the length specified.
   EXPECT_EQ(off, binary_data_size);
-  ASSERT_EQ(set.NumDescriptors(), 2U);
-  ASSERT_EQ(set.GetDescriptorRef(0).address, (dw_addr_t)0x1000);
-  ASSERT_EQ(set.GetDescriptorRef(0).length, (dw_addr_t)0x100);
-  ASSERT_EQ(set.GetDescriptorRef(1).address, (dw_addr_t)0x2000);
-  ASSERT_EQ(set.GetDescriptorRef(1).length, (dw_addr_t)0x10);
+  ASSERT_EQ(set.getDescriptorsSize(), 4U);
```

Comment:
This could be written in a number of ways that does not require these functions. My favourite is (requires adding an `operator==` to the Descriptor class):
```
ASSERT_THAT(set.descriptors(), testing::ElementsAre(Descriptor{0x1000, 0x100}, Descriptor{0x2000, 0x10}));
```

Reply:
Switched to use `DWARFDebugAranges` in this test. I think adding `operator==` is the same as using `getDescriptiorRef` and `getDescriptorsSize` because `operator==` is only used in the lldb test.

Reply:
The difference I see is that the name `operator==` does not reek of lldb, and is also something I can imagine being used in llvm (in tests or maybe even in production code). That said, I think that going for the higher level API is a good idea here. With the deletion of lldb class, I was not sure if we should even have those tests here, as this was something that should theoretically be tested on the llvm side.

Reply:
There's actually already a unit test for `llvm::DWARFArangeSet`. So, it makes sense to only test `lldb::DWARFAranges` here. I'll add those removed `lldb::DWARFArangeSet` tests to `llvm::DWARFArangeSet` unit test in another change.

Reply:
After checking https://github.com/llvm/llvm-project/blob/main/llvm/unittests/DebugInfo/DWARF/DWARFDebugArangeSetTest.cpp, I found it already covers those tests used here.